### ISSUE TITLE
feat request: plot utility to annotate figure with platform ID

### ISSUE THREAD

**Issue by eleanorfrajka** [callumrollo/glidertest#143]
For each plotting function which generates its own `fig`, add some annotation outside the axes (top left? bottom right?)  to print the platform ID from the `ds` input.

For example, GMT has a feature to add a little time stamp to figures:

![](https://docs.generic-mapping-tools.org/6.3/_images/GMT_Defaults_1c.png)

Could instead be the platform ID (logo optional).

**Comment by MOchiara** [comment]
I can make a function but we have a small problem at the moment. The relevant info are at the moment called `glider_serial` or `glider_model` but in OG1 they are supposed to be called with 'PLATFORM' at the beginning
![image](https://github.com/user-attachments/assets/8bfb14e4-b761-43e1-9d36-a645659d2ce7)

**Comment by MOchiara** [comment]
I a have a draft of this function (we have to fix the nomenclature in the attributes though). It is somewhat hard to choose a universal right location if we have it as a True or False (ex. add_annotation=True as a parameter in all plotting functions)

**Comment by callumrollo** [comment]
Yeah that's on me! A fault in the demo file. Will fix next week after the vocab/format meeting. I don't know how you've designed you draft, but for me this would make sense as a secondary function. Like:

```python
fig, ax = plots.plot_ts(ds)
fig = add_meta_to_fig(fig, ds, location='bottom right')
```

**Comment by MOchiara** [comment]
Yeah that of within the function and it is True as default and can be turned off but then it is annoying to position if you do want it but just not in that exact corner

**Comment by callumrollo** [comment]
Yeah, I think this works better as a secondary/wrapper function. That's why we return fig and ax after all! 

In my experience, once a function has more than ~ 4 arguments/kwargs it becomes real messy to maintain. So best to separate out the functionality

**Comment by eleanorfrajka** [comment]
I am less worried about whether it shows up in the right corner.  I kind of see it as a useful badge for non-production figures, and assume that if anyone wants a figure for an actual publication, that they'll pull out the `fig, ax` themselves for fine tuning.  (Or in a pinch, they'll just crop the tag, as long as it appears outside what would otherwise be necessary to include for a rectangular crop of the figure--retaining all axes labels, legends, colobars but squeezing off any excess whitespace.. the tag could then be in the whitespace.)
